lib/fileUploader/uploadManager.js

```diff
@@ -0,0 +1,107 @@
+const AzureUploader = require("./AzureUploader");
+const S3Uploader = require("./S3Uploader");
+const memoize = require('lodash.memoize');
+
+/**
+ * Handles file upload to Azure Blob Storage or AWS S3 based on the configuration
+ * by filepath.
+ * @param {string} filepath - The filepath to be uploaded.
+ * @param {string} runId - The ID of the run or identifier for the file.
+ */
+const handleFileUploadByPath = async (runId, filepath) => {
```

Comment:
Please transform into class

Reply:
in my opinion, this is not a good idea. It is better to use the uploadManager to handle different download options such as S3 or Azure